File: tests/fixtures/resend/resend-batch-size-not-enforced-broken/02-mapped-array.ts
import { Resend } from 'resend';

const resend = new Resend(process.env.RESEND_API_KEY);

// Builds a batch from a mapped array but never checks its length.
export async function sendReceipts(orders: { email: string }[]) {
  const messages = orders.map((o) => ({
    from: 'Acme <receipts@acme.com>',
    to: [o.email],
    subject: 'Receipt',
    html: '<p>Receipt</p>',
  }));

  return resend.batch.send(messages);
}

File: tests/fixtures/resend/resend-batch-size-not-enforced-broken/02-mapped-array.py
import os
import resend

resend.api_key = os.environ["RESEND_API_KEY"]


def send_receipts(orders):
    messages = [
        {
            "from": "Acme <receipts@acme.com>",
            "to": [o["email"]],
            "subject": "Receipt",
            "html": "<p>Receipt</p>",
        }
        for o in orders
    ]
    # Variable array with no len() guard — should flag.
    return resend.Batch.send(messages)
